tests/components/mqtt/test_init.py

```diff
@@ -67,51 +94,44 @@ def sensor_platforms_only():
 
 
 @pytest.fixture(autouse=True)
-def mock_storage(hass_storage):
+def mock_storage(hass_storage: dict[str, Any]) -> None:
     """Autouse hass_storage for the TestCase tests."""
 
 
 @pytest.fixture
-def device_reg(hass):
+def device_reg(hass: HomeAssistant) -> dr.DeviceRegistry:
     """Return an empty, loaded, registry."""
     return mock_device_registry(hass)
 
 
 @pytest.fixture
-def entity_reg(hass):
+def entity_reg(hass: HomeAssistant) -> er.EntityRegistry:
     """Return an empty, loaded, registry."""
     return mock_registry(hass)
 
 
 @pytest.fixture
-def mock_mqtt():
-    """Make sure connection is established."""
-    with patch("homeassistant.components.mqtt.MQTT") as mock_mqtt:
-        mock_mqtt.return_value.async_connect = AsyncMock(return_value=True)
-        mock_mqtt.return_value.async_disconnect = AsyncMock(return_value=True)
-        yield mock_mqtt
-
-
-@pytest.fixture
```

Comment:
This mock is not needed any more it can be replaced by awaiting `mqtt_mock_entry_no_yaml_config` or `mqtt_mock_entry_with_yaml_config`